src/main/resources/api/data-repository-openapi.yaml

```diff
@@ -969,6 +969,105 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/ErrorModel'
+  /api/repository/v1/snapshots/{id}/index:
+    post:
+      tags:
+        - snapshots
+        - repository
+      description: Create a new search index for a snapshot
+      operationId: createSearchIndex
+      parameters:
+        - name: id
+          in: path
+          description: A UUID to used to identify an object in the repository
+          required: true
+          schema:
+            type: string
+      requestBody:
+        description: Index to create from SQL
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/SearchIndexRequest'
+        required: true
+      responses:
+        200:
+          description: Return index result
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/SearchIndexModel'
+        400:
+          description: Bad request - invalid id, badly formed
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorModel'
+        403:
+          description: No permission to see snapshot
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorModel'
+        404:
+          description: Not found - snapshot id does not exist
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorModel'
+  /api/repository/v1/snapshots/query:
+    post:
+      tags:
+        - snapshots
+        - repository
+      description: Search against one or more snapshots
+      operationId: querySearchIndices
+      parameters:
+        - name: offset
+          in: query
+          description: The number of results to skip before retrieving the next
+            page
+          schema:
+            type: integer
+            default: 0
+        - name: limit
+          in: query
+          description: The number of search results to return
+          schema:
+            type: integer
+            default: 1000
```

Comment:
This may not be immediately useful. Should I remove it?

Reply:
I think it's OK to leave it in for now.